Link categories in threadlist on homepage

digitalcraftsman opened the pull request:
This makes it easier for users to explore threads of category.

jyapayne commented:
@digitalcraftsman Looks good, but the tests are failing for you because you need to modify any selectors in `tests/browsertests/categories.nim` from `.category-name` to `.category-name a`

digitalcraftsman commented:
Thanks for your feedback Joey. I've already had a brief look at the test cases but hadn't the time so far to fix them -- hence the draft status of this PR. I'll try to make the PR mergable tomorrow.

digitalcraftsman commented:
> the tests are failing for you because you need to modify any selectors in tests/browsertests/categories.nim from .category-name to .category-name a

I've changed every occurrence of `.category-name` to `.category-name a` and rebased this PR as well. Though the test are still failing. The information in the build log make it a bit hard to pin point the exact location of the error. Any ideas?

dom96 commented:
View Raw Logs -> https://pipelines.actions.githubusercontent.com/TfFc2dwZMJrm13lyYYjnBB0QevJUgsIMgYuB8Jk4w850uUEjef/_apis/pipelines/1/runs/45/signedlogcontent/2?urlExpires=2021-11-29T21%3A37%3A19.0099325Z&urlSigningMethod=HMACV1&urlSignature=Lv2mEWArKmCPCvVLaXHy37hXzc%2BE%2BdO6f81oLMadsrs%3D

Then search for `[FAIL`

dom96 commented:
Looking at our other runs it seems there is some flakiness :/ https://github.com/nim-lang/nimforum/actions

I've restarted this build.
